encoding/thrift/outbound.go

```diff
@@ -184,7 +155,7 @@ func (c thriftClient) Call(ctx context.Context, reqMeta yarpc.CallReqMeta, reqBo
 		return wire.Value{}, nil, err
 	}
 
-	envelope, err := proto.DecodeEnveloped(bytes.NewReader(payload))
+	envelope, err := call.Protocol.DecodeEnveloped(bytes.NewReader(payload))
```

Comment:
WithRequest returns the response with the raw bytes. We need to deserialize
those bytes in the client implementation somewhere.